assets/js/components/GatheringDataNotice.js

```diff
@@ -19,25 +19,32 @@
 /**
  * External dependencies
  */
+import classnames from 'classnames';
 import PropTypes from 'prop-types';
 
 /**
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
 
-function GatheringDataNotice( { style = 'default' } ) {
+function GatheringDataNotice( { style } ) {
 	return (
 		<div
-			className={ `googlesitekit-gathering-data-notice googlesitekit-gathering-data-notice--has-style-${ style }` }
+			className={ classnames( 'googlesitekit-gathering-data-notice', {
+				'googlesitekit-gathering-data-notice--has-style': !! style,
+				'googlesitekit-gathering-data-notice--has-style-default':
+					style === 'default',
+				'googlesitekit-gathering-data-notice--has-style-overlay':
+					style === 'overlay',
+			} ) }
 		>
-			<p>{ __( 'Gathering data…', 'google-site-kit' ) }</p>
+			<div>{ __( 'Gathering data…', 'google-site-kit' ) }</div>
```

Comment:
True, span did feel semantically a bit more correct to me too - but since it needed to be treated like a block and didn't contain any other inline elements, I leaned towards a div. But I'll change it to span.